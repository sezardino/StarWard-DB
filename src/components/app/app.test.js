import React from 'react';
import {shallow} from 'enzyme';

import App from './app';
describe('Component App', () => {
	const app = shallow(<App />);

	it('Component App rendered correctly', () => {
		expect(app.debug()).toMatchSnapshot();
	});

	it('Have initial state isLoggedIn false', () => {
		expect(app.state().isLoggedIn).toBeFalsy();
	});

	it('Handler logInHandler change state', () => {
		app.instance().logInHandler();
		expect(app.state().isLoggedIn).toBeTruthy();
		app.instance().logInHandler();
		expect(app.state().isLoggedIn).toBeFalsy();
		app.instance().logInHandler();
		expect(app.state().isLoggedIn).toBeTruthy();
	});
});
