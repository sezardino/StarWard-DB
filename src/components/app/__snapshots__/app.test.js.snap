// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component App Component App rendered correctly 1`] = `
"<ErrorBoundary>
  <BrowserRouter>
    <Header />
    <Component />
    <Switch>
      <Route path=\\"/\\" exact={true} render={[Function: render]} />
      <Route path=\\"/people/:id?\\" exact={true} component={[Function: C]} />
      <Route path=\\"/planets\\" exact={true} component={[function]} />
      <Route path=\\"/starships\\" exact={true} component={[Function: C]} />
      <Route path=\\"/starships/:id\\" render={[Function: render]} />
      <Route path=\\"/login\\" exact={true} render={[Function: render]} />
      <Route path=\\"/secret\\" exact={true} render={[Function: render]} />
      <Route render={[Function: render]} />
    </Switch>
  </BrowserRouter>
</ErrorBoundary>"
`;
